### PR TITLE
Bootstrap with nuget

### PR DESCRIPTION
**Pull request by anurse**
Closer and closer!

/cc @Sridhar-MS @brthor @piotrpMSFT @davidfowl

### CODE REVIEW

File: scripts/build/restore-packages.ps1 [hunk 1]
```diff
@@ -17,13 +17,14 @@ else {
...
 
 # Restore packages
+# NOTE(anurse): I had to remove --quiet, because NuGet3 is too quiet when that's provided :(
 header "Restoring packages"
-& "$DnxRoot\dnu" restore "$RepoRoot\src" --quiet --runtime "$Rid" "$NoCacheArg"
-& "$DnxRoot\dnu" restore "$RepoRoot\test" --quiet --runtime "$Rid" "$NoCacheArg"
-& "$DnxRoot\dnu" restore "$RepoRoot\tools" --quiet --runtime "$Rid" "$NoCacheArg"
+dotnet restore "$RepoRoot\src" --runtime "$Rid"
+dotnet restore "$RepoRoot\test" --runtime "$Rid"
+dotnet restore "$RepoRoot\tools" --runtime "$Rid"
```
Comment (anurse) [inline scripts/build/restore-packages.ps1]: NuGet3 doesn't have a `--no-cache` option. Instead, we'll want to clear the cache before each build. Having said that, using the cache for now should be fine.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test this please.

**Comment by anurse** [comment]
The remaining issues are:

* [ ] (All Non-Windows builds) Bug in the CoreFX packages where non-Windows implementation packages have a dependency on `System.Security.Principal.Windows`. Status: Fixed and waiting on CoreFX build to finish /cc @ericstj .
* [ ] (All builds, but only Windows is getting that far :)) ILC.exe is not published properly to the output due to https://github.com/dotnet/cli/issues/964 . Status: NuGet fix required, it's been merged, waiting on NuGet build to finish /cc @emgarten .

**Comment by piotrpMSFT** [comment]
This is broken via https://github.com/NuGet/Home/issues/1418. @anurse is working on a workaround. Thanks for helping us find it, @ericstj!

**Comment by anurse** [comment]
@dotnet-bot test CentOS7.1 Debug Build please

**Comment by anurse** [comment]
@dotnet-bot test OSX Debug Build please

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT Debug Build please.

**Comment by anurse** [comment]
Green checkmarks!?! Wow! Progress!

**Comment by piotrpMSFT** [comment]
Moar PJamagrams?

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Debug Build please

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Release Build please

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Debug Build please

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Release Build please

**Comment by piotrpMSFT** [comment]
@dotnet-bot test OSX Release Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT Debug Build please.

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT Release Build please.

**Comment by anurse** [comment]
These Windows failures seem very transient. I'll give them a quick look but might I suggest we just merge? @piotrpMSFT

**Comment by anurse** [comment]
@dotnet-bot test Windows_NT Release Build please

**Comment by piotrpMSFT** [comment]
@anurse, can you rebase? I was about to do it for you, but it's your PR :)

**Comment by anurse** [comment]
Will do

**Comment by anurse** [comment]
Rebased! Obviously I'd suggest we wait for at least some builds to pass to make sure I didn't bork up the merge :)

**Comment by piotrpMSFT** [comment]
if it looks as clean as this morning, let's open issues to track the remaining test failures, merge, and then fix the tests. Once it's merged we can help with cleanup...

**Comment by anurse** [comment]
image

**Comment by piotrpMSFT** [comment]
@dotnet-bot test Windows_NT Debug build please.

**Comment by piotrpMSFT** [comment]
I built Windows Debug on my local machine. corehost.exe is crashing during some tests. Checking it out...

**Comment by anurse** [comment]
Hmm.. that sounds non-ideal :)

**Comment by anurse** [comment]
Closing this as #1004 supersedes it.

**Pull request closed without merging**